lib/mumukit/bridge/runner/response_type.rb

```diff
@@ -39,10 +39,18 @@ def build_hash(response)
       private
 
       def parse_test_results(results)
-        results.map { |it| {
-            title: it['title'],
-            status: it['status'].to_sym,
-            result: it['result']} }
+        results.map do |it|
+          { summary: safe_compact(it['summary'])&.symbolize_keys.presence }
```

Comment:
Maybe: 
``` 
     {summary: it['summary']}.deep_compact&.symbolize_keys
     .merge(...)
```

and moving that to core?

Reply:
I'd prefer to avoid adding more and more dependencies to this feature, since it already involves mumukit, bridge, gobstones runner, domain and laboratory :disappointed: 

But I definitely can extract a `deep_compact` method and port it to core :smile:

Reply:
Well, I started to implement it, and I notice that what I need here is not a `deep_compact` but a new method that compacts presence of members  - maybe in a deep way. Something like this: 

```ruby
class Hash
  def prune
    transform_values { |it| it.prune.presence }.compact
  end
end

class Object
  def prune
    self
  end
end

class Array
  def prune
    map { |it| it.prune.presence }.compact
  end
end
```

However, in this case I don't really need a full-deep behavior, just one level of nesting, and then reject other ill formed hashes. 

So I am going to merge this as-is, and then PR this method to core.